wire-ios-notification-engine/Sources/NotificationSession.swift

```diff
@@ -409,18 +409,24 @@ extension NotificationSession: PushNotificationStrategyDelegate {
     private func processDecodedEvents(_ events: [ZMUpdateEvent]) {
         WireLogger.notifications.info("processing \(events.count) decoded events...")
 
+        // Dictionary to filter notifications fetched in same batch with same messageOnce
+        // i.e: textMessage and linkPreview
+        var tempNotifications = [Int: ZMLocalNotification]()
+
         for event in events {
             if let callEventPayload = callEventPayloadForCallKit(from: event) {
                 WireLogger.calling.info("detected a call event")
                 // Only store the last call event.
                 callEvent = callEventPayload
             } else if let notification = notification(from: event, in: context) {
                 WireLogger.notifications.info("generated a notification from an event")
-                localNotifications.append(notification)
+                tempNotifications[notification.contentHashValue] = notification
```

Comment:
I think it works this way as we need it. It feels a bit complex so I wondered if we can override the NSObject hash value and use a `Set<ZMLocalNotification>` over this dictionary? 
```swift
    public override var hash: Int {
        var hash = Hasher()
        hash.combine(messageNonce)
        hash.combine(title)
        hash.combine(body)
        return hash.finalize()
    }
```

Would it have other side effects maybe?

Reply:
it could be weird to have a custom implementation in other cases where we use ZMLocalNotification.